Dockerfile

```diff
@@ -0,0 +1,20 @@
+# Use an official Python runtime as a parent image
+FROM python:3.9-slim
+
+# Set the working directory in the container
+WORKDIR /app
+
+# Copy the current directory contents into the container at /app
+COPY . /app
+
+# Install any needed packages specified in requirements.txt
+RUN pip install --no-cache-dir -r requirements.txt
```

Comment:
Before this runs, gcc needs to be installed.

Otherwise this fails with the following error:

```
10.67 Building wheels for collected packages: llama-cpp-python
10.67   Building wheel for llama-cpp-python (pyproject.toml): started
10.79   Building wheel for llama-cpp-python (pyproject.toml): finished with status 'error'
10.79   error: subprocess-exited-with-error
10.79   
10.79   × Building wheel for llama-cpp-python (pyproject.toml) did not run successfully.
10.79   │ exit code: 1
10.79   ╰─> [16 lines of output]
10.79       *** scikit-build-core 0.10.7 using CMake 3.31.0 (wheel)
10.79       *** Configuring CMake...
10.79       loading initial cache file /tmp/tmpf4mbpzhc/build/CMakeInit.txt
10.79       CMake Error at /tmp/pip-build-env-aw4k351n/normal/lib/python3.9/site-packages/cmake/data/share/cmake-3.31/Modules/CMakeDetermineCCompiler.cmake:49 (message):
10.79         Could not find compiler set in environment variable CC:
10.79       
10.79         gcc.
10.79       Call Stack (most recent call first):
10.79         CMakeLists.txt:3 (project)
10.79       
10.79       
10.79       CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
10.79       CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
10.79       -- Configuring incomplete, errors occurred!
10.79       
10.79       *** CMake configuration failed
10.79       [end of output]
10.79   
10.79   note: This error originates from a subprocess, and is likely not a problem with pip.
10.79   ERROR: Failed building wheel for llama-cpp-python
10.79 Failed to build llama-cpp-python
10.79 ERROR: Could not build wheels for llama-cpp-python, which is required to install pyproject.toml-based projects
10.94 
10.94 [notice] A new release of pip is available: 23.0.1 -> 24.3.1
10.94 [notice] To update, run: pip install --upgrade pip
------
failed to solve: process "/bin/sh -c pip install --no-cache-dir -r requirements.txt" did not complete successfully: exit code: 1
```

Reply:
Somewhat related to #16 

Also, forget to mention, but its not just gcc, its both gcc and g++

Reply:
You need GCC installed anyways to install the requirements for the project at all. That isn't exclusive to this PR @ToasterUwU

Reply:
You'll notice it's failing to build llama-cpp-python, which isn't a requirement that this PR changed or added

Reply:
The only reason i linked that issue is for showing context of what is going wrong. Its the exact same issue (missing dependencies) but here its within the Docker container, so the exact same step is missing in the Dockerfile, just like that person who made that issue was missing that step on their own OS

Reply:
Oh I get it. Yeah, installing on base OS it's on the user to have gcc and g++ installed, but the docker container should handle that. Makes perfect sense. Sorry for the confusion @ToasterUwU